pkg/isoeditor/nmstate_handler.go

```diff
@@ -31,24 +32,62 @@ func NewNmstateHandler(workDir string, executer Executer) NmstateHandler {
 }
 
 func (n *nmstateHandler) CreateNmstateRamDisk(rootfsPath, ramDiskPath, nmstatectlPath string) error {
-	compressedCpio, err := n.buildNmstateCpioArchive(rootfsPath)
+	//compressedCpio, err := n.ExtractAndCacheNmstatectl(rootfsPath, nmstatectlPath)
+	//if err != nil {
+	//	return err
+	//}
+	var compressedCpio []byte
+	var readErr error
+
+	if _, err := os.Stat(nmstatectlPath); err == nil {
+		// Read and return the cached content
+		compressedCpio, readErr = os.ReadFile(nmstatectlPath)
+		if readErr != nil {
+			return fmt.Errorf("failed to read cached nmstatectl: %v", readErr)
+		}
+	} else if os.IsNotExist(err) {
+		// File doesn't exist - this should be an error condition
+		return fmt.Errorf("nmstatectl cache file not found: %s", nmstatectlPath)
+	} else {
+		// Some other error occurred
+		return fmt.Errorf("failed to stat nmstatectl cache file: %v", err)
+	}
+
+	// write for RAM disk
+	err := os.WriteFile(ramDiskPath, compressedCpio, 0755) //nolint:gosec
 	if err != nil {
 		return err
 	}
 
-	// write for RAM disk
-	err = os.WriteFile(ramDiskPath, compressedCpio, 0755) //nolint:gosec
+	return nil
+}
+
+func (n *nmstateHandler) ExtractAndCacheNmstatectl(rootfsPath, nmstatectlPath string) error {
+	// Check if nmstatectl is already cached
+	//if _, err := os.Stat(nmstatectlPath); err == nil {
+	//	log.Debugf("nmstatectl already cached at %s", nmstatectlPath)
+	//	// Read and return the cached content
+	//	cachedContent, readErr := os.ReadFile(nmstatectlPath)
+	//	if readErr != nil {
+	//		return nil, fmt.Errorf("failed to read cached nmstatectl: %v", readErr)
+	//	}
+	//	return cachedContent, nil
+	//}
+
+	log.Debugf("Extracting and caching nmstatectl from %s to %s", rootfsPath, nmstatectlPath)
+
+	compressedCpio, err := n.buildNmstateCpioArchive(rootfsPath)
```

Comment:
How about we make this the public method here and move the write into the caller?